apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -223,60 +237,40 @@ public override void ActivateOptions()
                         };
                         break;
                     case PlatformType.Unknown:
-                        GaxPreconditions.CheckState(!string.IsNullOrEmpty(ProjectId), $"{nameof(ProjectId)} must be set on this platform.");
-                        platformProjectId = ProjectId;
-                        platformResource = new MonitoredResource
-                        {
-                            Type = "global",
-                            Labels =
-                            {
-                                {"project_id", ProjectId }
-                            }
-                        };
                         break;
                     default:
                         throw new InvalidOperationException($"Unrecognized platform detected: '{platform.Type}'");
                 }
-                _logName = new LogName(PreferPlatformResourceTypeDetection ? platformProjectId : (ProjectId ?? platformProjectId), LogId).ToString();
-                _resource = PreferPlatformResourceTypeDetection ? platformResource : (ResourceType != null ? new MonitoredResource
-                {
-                    Type = ResourceType,
-                    Labels = { _resourceLabels.ToDictionary(x => x.Key, x => x.Value) }
-                } : platformResource);
             }
-            switch (LocalQueueType)
+            if (projectId == null)
             {
-                case LocalQueueType.Memory:
-                    _logQ = new MemoryLogQueue(MaxMemorySize, MaxMemoryCount);
-                    break;
-                case LocalQueueType.Disk:
-                    throw new NotImplementedException("File-base local queues not implemented.");
-                default:
-                    throw new InvalidOperationException("Inconceivable!");
+                // Either platform detection is disabled, or it detected an unknown platform
+                projectId = ProjectId;
+                if (ResourceType == null)
+                {
+                    resource = new MonitoredResource { Type = "global", Labels = { { "project_id", projectId } } };
+                }
+                else
+                {
+                    resource = new MonitoredResource { Type = ResourceType, Labels = { _resourceLabels.ToDictionary(x => x.Key, x => x.Value) } };
+                }
             }
-            _initIdTask = Task.Run(_logQ.GetPreviousExecutionIdAsync);
-            var labels = new Dictionary<string, string>();
-            foreach (var customLabel in _customLabels)
+            else
             {
-                labels.Add(customLabel.Key, customLabel.Value);
+                if (ProjectId != null && projectId != ProjectId)
+                {
+                    // TODO: Warning
+                    preLogs.Add(() => BuildLogEntry(new LoggingEvent(new LoggingEventData
```

Comment:
At the time BuildLogEntry is called, _logName is null?  Should this be moved to after _logName, _resource is set?